UnitTests/Off/boxcube/src/boxcube.cpp

```diff
@@ -0,0 +1,186 @@
+/*
+        Copyright (c) 2021 Technical University of Munich
+	Chair of Computational Modeling and Simulation.
+
+	TUM Open Infra Platform is free software; you can redistribute it and/or modify
+	it under the terms of the GNU General Public License Version 3
+	as published by the Free Software Foundation.
+
+	TUM Open Infra Platform is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <namespace.h>
+
+#include <OffVisualTest.h>
+
+#include <OffConverter/OffReader.h>
+
+
+using namespace testing;
+
+
+class Boxcube : public OffVisualTest
+{
+protected:
+
+	// Test standard values
+	buw::Image4b _background = buw::Image4b(0, 0);
+
+	virtual void SetUp() override {
+		OffVisualTest::SetUp();
+
+		model = oip::OffReader::readFile(filename.string());
+
+		_background = rendererOff->captureImage();
+                rendererOff->setModel(model);
+	}
+
+	virtual void TearDown() override {
+		model.reset();
+		OffVisualTest::TearDown();
+	}
+
+	virtual std::string TestName() const { return "boxcube"; }
+
+	const boost::filesystem::path filename = dataPath("boxcube.off");
+
+	buw::ReferenceCounted<oip::OffModel> model = buw::makeReferenceCounted<oip::OffModel>();
+};
+
+TEST_F(Boxcube, AllEdgesAndFacesRed)
+{
+	// check for the number of vertices 
+	// does not coincide with the number of vertices mentioned in the off file since some vertices are
+	// stored once for every triangle they are part of
+	EXPECT_THAT(model->geometry().vertices.size(), Eq(324));
+	// check for the number of indices (see comment for vertices)
+	EXPECT_THAT(model->geometry().indices.size(), Eq(324));
```

Comment:
Are you sure about the numbers here?
The file defines 48 vertices so I wonder how we end up with 324

Reply:
I made it because when debugging it failed the Test. And actually the value of this Test producing in console (see scr)
![image](https://user-images.githubusercontent.com/76201952/142609552-b1f23042-57f7-4801-8b46-e0ead6e0763b.png)